gammapy/datasets/map.py

```diff
@@ -586,10 +586,8 @@ def meta(self, value):
     # TODO: keep or remove?
     @property
     def background_model(self):
-        try:
-            return self.models[f"{self.name}-bkg"]
-        except (ValueError, TypeError):
-            pass
+        if self.models and self.name in self.models.background_models.keys():
+            return self.models[self.models.background_models[self.name]]
```

Comment:
if the condition is not satisfied, we return nothing? Is it the expected behaviour?

Reply:
yes should return None